lib/mix/test/mix/tasks/format_test.exs

```diff
@@ -167,6 +167,20 @@ defmodule Mix.Tasks.FormatTest do
     assert output == ""
   end
 
+  test "checks that no error is raised with --check-formatted and --no-exit" do
+    capture_io("foo( )", fn ->
+      Mix.Tasks.Format.run(["--check-formatted", "--no-exit", "-"])
+    end)
+
+    message =
+      receive do
+        {:mix_shell, :info, [message]} ->
+          message
+      end
+
+    assert message =~ "The following files are not formatted"
```

Comment:
@josevalim This feels hacky to me, but since you can't pattern on part of a string with `assert_received`, I couldn't think of another way to do this. 😅